### DISCUSSION TITLE
Things about c0d3.com that make students frustrated

### DISCUSSION THREAD

**Discussion by anthonykhoa** [garageScript/c0d3-app#1267] [Ideas]
* Recursion
  * Not being able to comment on reviews -- it should be easier to have a conversation with the reviewer
    * Add `chat with me` button to comment section of challenge review 
  * Students disagree with reviewer
  * Email problem
  * Allow anonymous users to view the curriculum
  * Improve CLI -- better detection against bad submissions, no for/while loops
  * Git problems -- make git games!
  * Are you a windows user? link to chatroom to windows chatroom
  * Host lesson docs on c0d3 instead of notion
  * after js3, people stop js4, classes are hard in js4, section on OOP
  * setting up webpack/babel, needs more info on it
  * module on tests
  * change messaging on landing page, make more appealing
  * async await, and testing with async await

**Comment by ggwadera** [comment]
> - change messaging on landing page, make more appealing

For this I think we should highlight on the landing page that c0d3 was/is built entirely by it's students.

Maybe also put some testimonials/feedback from people who got into coding because of c0d3.com.

**Comment by Ulisseus** [comment]
Quite a few students submit multiply files for js1 challenges because they didn't merge branches after js0 properly. I think that these git sections should be more clear. Currently lesson0 has step-by-step guide for updating master after completing submissions and lesson1 has instructions for updating challenge branches. 

I propose to move everything to the end of lesson1:
 * Add pictures with git graph along with showing `git log --graph --decorate`. 
 * Specify that if everything went correctly `git diff master` from challenge branch should be empty. 
 * Explain that submitting challenge means sending `git diff master` string to our server, so users can check what exactly they're going to submit before they actually hit the button. 
 * Maybe mention a few git guis (git kraken and etc). 
 
 Not sure about the last one, learning cli commands properly is still the best way.

**Comment by ggwadera** [comment]
For the git guis I like using the one built-in on VS Code, it's pretty simple and gets the job done.

The git lens extension is also very helpful as it expands the gui capabilities to also manage branches, remotes, etcetera.
